docuvision/batchsendfulldir.py

```diff
@@ -0,0 +1,204 @@
+    default=".", type=str)
+ap.add_argument("--loglevel", help="Logging level. Options are DEBUG, INFO, WARNING, ERROR, CRITICAL.", 
+    default="DEBUG", type=str)
+
+args, unknown = ap.parse_known_args()
+if args.dir == '.': args.dir=os.path.dirname(os.path.realpath(__file__))
+print("DOCUVISION SCRIPT STARTED")
+print("args = ", args)
+
+#setup logging
+logging.basicConfig(
+    filename='docuvision.log', 
+    filemode='a', 
+    level=args.loglevel.upper(),
+    format='%(asctime)s:%(levelname)s:%(message)s')
+logging.info("STARTING NEW JOB. {}".format(datetime.datetime.now()))
+logging.info("Processing {} filetypes in {} ...".format(args.types, args.dir))
+
+#%% SETUP AUTH
+#make sure token and address are set in env vars (will use the default address here if not)
+APIADDRESS = os.environ.get('DOCUVISION_API_ADDRESS', 
+    "https://services.hank.ai/docuvision/v1/tasks/") #PROD
+    #"https://services-dev.hank.ai/docuvision/v1/tasks/") #DEV
+
+APITOKEN = os.environ.get('DOCUVISION_API_TOKEN', None)
+if APITOKEN is None:
+    logging.error("DOCUVISION_API_TOKEN environment variable is not set. Aborting")
+    sys.exit()
+if APIADDRESS is None:
+    logging.error("DOCUVISION_API_ADDRESS environment variable is not set. Aborting")
+    sys.exit()
+if APIADDRESS[-1] != "/": APIADDRESS+="/"
+logging.info("APIADDRESS loaded. Using {}".format(APIADDRESS))
+logging.info("APITOKEN loaded. Using {}...".format(APITOKEN[:15]))
+
+
+#expects filepath to be a Path() object
+#timeout is in seconds
+#writes the newly created 'jobid filepath' as a line in pendingjobs.docuvision file in current directory
+#returns job id (int)
+def hankai_submit_job(filepath, timeout=60):
+    with open(file, 'rb') as f:
+        raw_bytes = f.read()
+        enc_bytes = base64.encodebytes(raw_bytes)
+    enc_string = enc_bytes.decode("utf-8")
+    md5_sum = hashlib.md5(raw_bytes).hexdigest()
+    headers = {"x-api-key": APITOKEN }
+    req = {
+        "name": "customername_job_x",
+        "request": {
+            "document": {
+                "name": filepath.name,
+                "dataType": "blob",
+                "encodingType": "base64/utf-8",
+                "mimeType": filepath.suffix[1:], #removes the leading period
+                "model": "base-medrec-anesthesia", #modify this to the model you want to consume
+                "confidenceInterval": 0.7, #modify this to your liking
+                "isPerformOCR": True, #if set your result will have OCR'd words and bounding boxes
+                "sizeBytes": len(enc_bytes),
+                "md5Sum": md5_sum,
+                "data": enc_string,
+            }
+        }
+    }
+    resp = requests.request(
+        method="POST",
+        url=f"{APIADDRESS}", #docuvision/v1/tasks/7",
+        headers=headers,
+        data=json.dumps(req),
+        timeout=timeout
+    )
+    rjson = json.loads(resp.content)
+    print(resp.content)
```

Comment:
Is it worth printing the content? Seems like a degug function. Likely get noisy in production

Reply:
meant to take it out